src/EditorFeatures/Core/DocumentationComments/CopilotGenerateDocumentationCommentProvider.cs

```diff
@@ -0,0 +1,333 @@
+{
+    internal class CopilotGenerateDocumentationCommentProvider : SuggestionProviderBase
+    {
+        private SuggestionManagerBase? _suggestionManagerBase;
+        private VisualStudio.Threading.IAsyncDisposable? _intellicodeLineCompletionsDisposable;
+
+        internal SuggestionSessionBase? _suggestionSession;
+
+        public readonly IThreadingContext ThreadingContext;
+
+        public CopilotGenerateDocumentationCommentProvider(IThreadingContext threadingContext)
+        {
+            ThreadingContext = threadingContext;
+        }
+
+        public async Task InitializeAsync(ITextView textView, SuggestionServiceBase suggestionServiceBase, CancellationToken cancellationToken)
+        {
+            _suggestionManagerBase = await suggestionServiceBase.TryRegisterProviderAsync(this, textView, "AmbientAIDocumentationComments", cancellationToken).ConfigureAwait(false);
+        }
+
+        public async Task<ICopilotCodeAnalysisService?> IsGenerateDocumentationAvailableAsync(Document document, CancellationToken cancellationToken)
+        {
+            // Bailing out if copilot is not available or the option is not enabled.
+            if (document.GetLanguageService<ICopilotCodeAnalysisService>() is not { } copilotService ||
+                    await copilotService.IsAvailableAsync(cancellationToken).ConfigureAwait(false) is false)
+            {
+                return null;
+            }
+
+            if (document.GetLanguageService<ICopilotOptionsService>() is not { } copilotOptionService ||
+                !await copilotOptionService.IsGenerateDocumentationCommentOptionEnabledAsync().ConfigureAwait(false))
+            {
+                return null;
+            }
+
+            if (_suggestionManagerBase is null)
+            {
+                return null;
+            }
+
+            return copilotService;
+        }
+
+        public async Task GenerateDocumentationProposalAsync(Document document, DocumentationCommentSnippet snippet,
+        ITextSnapshot oldSnapshot, VirtualSnapshotPoint oldCaret, CancellationToken cancellationToken)
+        {
+            await Task.Yield().ConfigureAwait(false);
+
+            var copilotService = await IsGenerateDocumentationAvailableAsync(document, cancellationToken).ConfigureAwait(false);
+
+            if (copilotService is null)
+            {
+                return;
+            }
+
+            var snippetProposal = GetSnippetProposal(snippet.SnippetText, snippet.MemberNode, snippet.Position, snippet.CaretOffset);
+
+            if (snippetProposal is null)
+            {
+                return;
+            }
+
+            // Do not do IntelliCode line completions if we're about to generate a documentation comment
+            // so that won't have interfering grey text.
+            _intellicodeLineCompletionsDisposable = await _suggestionManagerBase!.DisableProviderAsync(SuggestionServiceNames.IntelliCodeLineCompletions, cancellationToken).ConfigureAwait(false);
```

Comment:
I wonder if we should move both `_intellicodeLineCompletionsDisposable` and `_suggestionSession` to `DocumentationCommentSuggestion`, since they have the same life-cycle. This way once a suggestion is passed to session for display, it no longer needs to interact with the provider